src/OneinchCallee.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2023 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+pragma experimental ABIEncoderV2;
+
+interface GemJoinLike {
+    function dec() external view returns (uint256);
+
+    function gem() external view returns (TokenLike);
+
+    function exit(address, uint256) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external;
+
+    function transfer(address, uint256) external;
+
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface IAggregationExecutor {
+    function callBytes(address msgSender, bytes calldata data) external payable;
+}
+
+interface OneinchRouter {
+    struct SwapDescription {
+        TokenLike srcToken;
+        TokenLike dstToken;
+        address srcReceiver;
+        address dstReceiver;
+        uint256 amount;
+        uint256 minReturnAmount;
+        uint256 flags;
+    }
+
+    function swap(
+        IAggregationExecutor executor,
+        SwapDescription calldata desc,
+        bytes calldata permit,
+        bytes calldata data
+    ) external payable returns (uint256 returnAmount, uint256 spentAmount);
+}
+
+contract OneinchCallee {
+    DaiJoinLike public daiJoin;
+    TokenLike public dai;
+    uint256 public constant RAY = 10**27;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, 'ds-math-add-overflow');
+    }
+
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, 'ds-math-sub-underflow');
+    }
+
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _add(x, _sub(y, 1)) / y;
+    }
+
+    constructor(address daiJoinAddress) public {
+        daiJoin = DaiJoinLike(daiJoinAddress);
+        dai = daiJoin.dai();
+        dai.approve(daiJoinAddress, 2**256 - 1);
```

Comment:
convention when setting variables in the contract that need to be passed in as function args is to either prefix or suffix them with an `_`.  This repo already suffixes, so I changed the convention to that. 
 I also stripped the `Address` from the variable names for the same reason.  Lastly, I changed the convention for max uint, usually expressed in `0.7.x` and later as `type.max(uint256)`, to the `0.6.12` equivalent `uint256(-1)`.  You had trouble compiling this because foundry was just told to use the latest version, where dapp is hard locked to `0.6.12`.  I think we should stick with dapp in this PR, but this entire repo should be migrated over to foundry, which we can do in another PR.
```suggestion
    constructor(address daiJoin_) public {
        daiJoin = DaiJoinLike(daiJoin_);
        dai = daiJoin.dai();
        dai.approve(daiJoin_, uint256(-1));
```

Reply:
Actually foundry also compiles it with 0.6.12 according to the logs, it's only the linter which didn't pick it up, I've set it to the correct version. 

But can you explain why `uint256(-1)` is preferred over `2**256 - 1`? I though they are interchangeable, but `2**256 - 1` does not depend on the value overflow, according to [this discussion](https://forum.openzeppelin.com/t/using-the-maximum-integer-in-solidity/3000)

Reply:
As [per another suggestion](https://github.com/makerdao/exchange-callees/pull/34#discussion_r1070526673), this was changed to `type(uint256).max`